src/main/java/uk/gov/mca/beacons/service/model/Beacon.java

```diff
@@ -1,43 +1,67 @@
 package uk.gov.mca.beacons.service.model;
 
-import java.time.LocalDate;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import java.time.LocalDateTime;
+import java.util.List;
 import java.util.UUID;
 import javax.persistence.Entity;
 import javax.persistence.EntityListeners;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
 import javax.persistence.GeneratedValue;
 import javax.persistence.Id;
-import javax.persistence.Table;
+import javax.persistence.Transient;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
 import org.springframework.data.annotation.CreatedDate;
-import org.springframework.data.annotation.LastModifiedDate;
 import org.springframework.data.jpa.domain.support.AuditingEntityListener;
 
 @Entity
 @EntityListeners(AuditingEntityListener.class)
-@Table(name = "beacons")
 public class Beacon {
 
   @Id
   @GeneratedValue
   private UUID id;
 
-  private String beaconType;
+  @Enumerated(EnumType.STRING)
+  private BeaconType beaconType;
+
+  @NotNull
   private String hexId;
+
+  @NotNull
   private String manufacturer;
+
+  @NotNull
   private String model;
-  private String serialNumber;
-  private LocalDate batteryExpiry;
-  private LocalDate lastServiced;
-  private String checksum;
-  private String beaconStatus;
-  private String coding;
-  private String protocolCode;
+
+  @NotNull
+  private String manufacturerSerialNumber;
+
+  private String chkCode;
+
+  private LocalDateTime batteryExpiryDate;
```

Comment:
Updated these to local date time as we are constructing date objects in the frontend and sending this to the backend.

The difference between `LocalDateTime` and `LocalDate` is that `LocalDateTime` stores year, month, day, time whereas `LocalDate` is just year, month, day.

If we need to present this as just year, month, day to the user in the frontend we can strip out the time as needed